spritecontrol.cpp

```diff
@@ -2,19 +2,23 @@
 #include "resources.h"
 #include "config.h"
 #include "noise.h"
+#include "configdialog.h"
 
 using std::get;
 
 SpriteGenerator::SpriteGenerator() {
 	using namespace std::chrono;
 	std::srand(duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count());
 
-	std::vector<int> bag_of_palettes(_PALETTE_COUNT);
-	std::iota(bag_of_palettes.begin(), bag_of_palettes.end(), 0);
+	//std::vector<int> bag_of_palettes(static_cast<int>(PaletteName::_PALETTE_COUNT));
+	//std::iota(bag_of_palettes.begin(), bag_of_palettes.end(), 0);
+	std::vector<PaletteName> bag_of_palettes = PALETTES_BY_GROUP.at((PaletteGroup) (cfg.at(YonkPalette)));
 
-	for (int i = 0; i < round(cfg.at(MaxColors)); i++) {
+	float maxColors = cfg.at(MaxColors) * (bag_of_palettes.size() / config_ranges.at(MaxColors).second);
```

Comment:
We really want `MaxColors` to be a value between 0.0 and 1.0, but we can't just do that because people's existing configs will be outside of that range. So I guess this is fine for now.

Reply:
However - styling!! `snake_case` for locals please. I'm pointing this out because VS won't enforce it, even with the new .editorconfig.